### PR TITLE
Add `data_type` to `open_data` method in `DataStore` class

### LINKED ISSUE #1030: Data store framework enhancement: add the optional argument `data_type` to the `open_data` method
**Is your feature request related to a problem? Please describe.**
If the user wants to open a data resource as a multi-level dataset, the user needs to define the `opener_id="mldataset:<format>:<protocol>"`, which is not user friendly.  

**Describe the solution you'd like**
 In most of the cases format and protocol will be defined by the data resource. Therefore, it would be user friendly to add the optional argument `data_type`, where the user can decide between `"mldataset"` and `"dataset"`. The `opener_id` is then built dynamically.

### PR DESCRIPTION
**Pull request by konstntokas**
Closes #1030

Checklist:

* [x] Add unit tests and/or doctests in docstrings
* [x] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] ~New/modified features documented in `docs/source/*`~
* [x] Changes documented in `CHANGES.md`
* [x] GitHub CI passes
* [x] AppVeyor CI passes
* [x] Test coverage remains or increases (target 100%)

### CODE REVIEW

**Comment by forman** [comment]
FYI @TonioF (xcube-cci), @pont-us (xcube-cds), @TejasMorbagal (xcube-cmems)

**Comment by konstntokas** [comment]
As suggested, I included in keyword argument `data_type` in `**open_params` now. Further I added testing to the `open_data` method in the `ReferenceDataStore` class. Please check if I understood correctly, that the `open_data` method in the `ReferenceDataStore` class only returns `xr.Dataset`. However most of these tests are skipped because kerchunk is not installed (see e.g. https://github.com/xcube-dev/xcube/blob/main/test/core/store/ref/test_store.py#L178). Should we maybe add kerchunk to the testing section in the `environment.yml` ?

File: test/core/gen2/local/test_usercode.py [hunk 1]
```diff
@@ -104,7 +104,7 @@ def test_class_bad_params(self):
             "\n"
             "Failed validating 'minimum'"
             " in schema['properties']['value']:\n"
-            "    {'minimum': 1, 'type': 'integer'}\n"
+            "    {'type': 'integer', 'minimum': 1}\n"
```
Comment (konstntokas) [inline test/core/gen2/local/test_usercode.py:107]: This test did not fail on my local machine, but did fail in the CI. Adjusting it, fixed the CI. Still I do not understand this behavior.

**Comment by forman** [comment]
> Please check if I understood correctly, that the `open_data` method in the `ReferenceDataStore` class only returns `xr.Dataset`. 

True.

> However most of these tests are skipped because kerchunk is not installed (see e.g. https://github.com/xcube-dev/xcube/blob/main/test/core/store/ref/test_store.py#L178). Should we maybe add kerchunk to the testing section in the `environment.yml` ?

Let's do that!

File: test/core/gen2/local/test_usercode.py (hunk 1 above)
Comment (forman) [reply test/core/gen2/local/test_usercode.py:107]: I guess, what is formatted here is not an ordered dict, so the order of fields may change from build to build or even run to run.

**Review by forman — APPROVED** [review]

**Pull request merged**